locale.c

```diff
@@ -6680,19 +6724,15 @@ Perl_my_strerror(pTHX_ const int errnum)
 
     SETLOCALE_UNLOCK;
 
-#  endif /* End of doesn't have strerror_l */
-
-    DEBUG_Lv((PerlIO_printf(Perl_debug_log,
-              "Strerror returned; saving a copy: '"),
-              print_bytes_for_locale(errstr, errstr + strlen(errstr), 0),
-              PerlIO_printf(Perl_debug_log, "'\n")));
-
-#endif   /* End of does have locale messages */
+    DEBUG_STRERROR_RETURN(errstr);
 
     SAVEFREEPV(errstr);
     return errstr;
 }
```

Comment:
> So this commit separates them out into 5 different
> functions, with the result that it's clear what is going on in each.

This commit splits out the ends of the function definitions but the start of the 5 different function definitions aren't split out until the next commit.

Reply:
revised